src/app/ui-components/sidebar-right/sidebar-right.component.ts

```diff
@@ -60,5 +69,42 @@ export class SideBarRightComponent implements OnInit, OnDestroy {
         this.selectedObservable?.subscribe((uuid?: string) => {
             this.router.navigate([uuid ?? "."], {relativeTo: this.route});
         });
+        this.voiceAssistantService.voiceAssistantStateObservable.subscribe(
+            (state: VoiceAssistantState) => {
+                this.voiceAssistantActivationToggle.setValue(state.turnedOn);
+            },
+        );
+    }
+
+    removeCssClass(uuid: string) {
+        const videoSettingsButton = document.getElementById(
+            "dropdownbutton-" + uuid,
+        );
+        videoSettingsButton?.classList.remove("showPopover");
+    }
+
+    addCssClass(uuid: string) {
+        const videoSettingsButton = document.getElementById(
+            "dropdownbutton-" + uuid,
+        );
+        videoSettingsButton?.classList.add("showPopover");
+    }
+
+    toggleVoiceAssistant() {
+        const turnedOn = !this.voiceAssistantActivationToggle.value;
+        const nextState: VoiceAssistantState = {turnedOn, chatId: ""};
+        if (turnedOn) {
+            const match = RegExp(
+                `/voice-assistant/${CerebraRegex.UUID}/chat/(${CerebraRegex.UUID})`,
+            ).exec(this.router.url);
+            if (match) nextState.chatId = match[1];
+            else throw new Error("no chat selected");
+        }
+
+        this.vaState = turnedOn;
+
+        this.voiceAssistantService.setVoiceAssistantState(nextState).subscribe({
+            error: (error) => console.error(error),
+        });
     }
```

Comment:
move this to voice-assistant-chat component (see comment in sidebar-right.component.ts)